### PR TITLE
Updated boards to include any missing ones

### PR DESCRIPTION
**Pull request by makermelissa**
New CircuitPython Boards
 * adafruit_feather_esp32s2_nopsram
 * adafruit_feather_esp32s2_tftback_nopsram
 * adafruit_funhouse
 * adafruit_itsybitsy_rp2040
 * adafruit_qtpy_rp2040
 * cp_sapling_m0_spiflash
 * neopixel_trinkey_m0
 * rotary_trinkey_m0
 * sparkfun_nrf52840_micromod
 * sparkfun_pro_micro_rp2040
 * sparkfun_thing_plus_rp2040
 * stackrduino_m0_pro
 * stm32f411ce_blackpill_with_flash
 * thunderpack_v12

Renamed Boards
* thunderpack -> thunderpack_v11

Removed CircuitPython Boards
 * bdmicro_vina_m0
 * pca10040

New Blinka Boards
 * Rock Pi 4C
 * BeagleBone

### CODE REVIEW

File: _board/ADM_B_NRF52840_1.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 layout: download
-board_id: "unknown"
+board_id: "ADM_B_NRF52840_1"
 title: "Unknown Board Download"
 name: "Unknown Board"
 manufacturer: "Unknown"
```
Comment (dhalbert) [inline _board/ADM_B_NRF52840_1.md:6]:
```suggestion
title: "ADM 52480 board for Holyiot_18010 module Download"
name: "ADM 52480 board for Holyiot_18010 module"
manufacturer: "AtelierDuMaker"
```

**Review by ladyada — DISMISSED** [review]
omg!

File: _board/adafruit_feather_esp32s2_tftback_nopsram.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+layout: download
+board_id: "adafruit_feather_esp32s2_tftback_nopsram"
+title: "Feather ESP32-S2 with TFT Download"
+name: "Feather ESP32-S2 with TFT "
```
Comment (dhalbert) [inline _board/adafruit_feather_esp32s2_tftback_nopsram.md:5]:
```suggestion
name: "Feather ESP32-S2 with TFT"
```

File: _board/stackrduino_m0_pro.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+---
+layout: download
+board_id: "stackrduino_m0_pro"
+title: "StackRduino M0+ PRO Download"
+name: "StackRduino M0+ PRO"
```
Comment (dhalbert) [inline _board/stackrduino_m0_pro.md:5]: You deleted the manufacturer from some other titles and names, so I'm doing that here.

```suggestion
title: "M0+ PRO Download"
name: "M0+ PRO"
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
Thanks! Just a few description changes I noticed.

File: _board/stackrduino_m0_pro.md (hunk 3 above)
Comment (makermelissa) [reply _board/stackrduino_m0_pro.md:5]: I left this one because the official board name has the manufacturer name in it.

Comment (dhalbert) [reply _board/stackrduino_m0_pro.md:5]: :+1:

**Review by dhalbert — APPROVED** [review]

**Comment by dglaude** [comment]
Not sure this is the right place... but on the support matrix ( https://circuitpython.readthedocs.io/en/latest/shared-bindings/support_matrix.html ) I noticed that QTPy has no space between QT and Py in the new RP2040 version:
"Adafruit QT Py M0" and "Adafruit QT Py M0 Haxpress" vs "Adafruit QTPy RP2040"

**Comment by dglaude** [comment]
If you know the right place (new issue? on circuitpython-org or elsewhere) for that kind of silly comment about the support matrix...
I found other that hurt my eyes/brain:

IMXRT1010-EVK vs iMX RT 1060 EVK

Arduino MKR Zero vs Arduino MKR1300 (maybe that is their name decision)

**Comment by Neradoc** [comment]
@dglaude The names come from `MICROPY_HW_BOARD_NAME` in each board's `mpconfigboard.h`.
It's also the name that is show in boot_out.txt.

https://github.com/adafruit/circuitpython/blob/e6aaf9924f6cf49c7ea8327f181960ecb2fd636c/ports/raspberrypi/boards/adafruit_qtpy_rp2040/mpconfigboard.h#L1

**Comment by dhalbert** [comment]
> IMXRT1010-EVK vs iMX RT 1060 EVK

The first is the manufacturer's part number for the board; the second is the name.

> Arduino MKR Zero vs Arduino MKR1300 (maybe that is their name decision)

Yes, this as as found on Arduino's website.

**Comment by makermelissa** [comment]
> Not sure this is the right place... but on the support matrix ( https://circuitpython.readthedocs.io/en/latest/shared-bindings/support_matrix.html ) I noticed that QTPy has no space between QT and Py in the new RP2040 version:
> "Adafruit QT Py M0" and "Adafruit QT Py M0 Haxpress" vs "Adafruit QTPy RP2040"

Yeah... some of these boards use the names interchangeably like MatrixPortal vs Matrix Portal, which has both printed on the same board.

**Pull request merged**